synapse/rest/admin/rooms.py

```diff
@@ -231,6 +231,30 @@ async def on_GET(self, request, room_id):
         return 200, ret
 
 
+class RoomMembersRestServlet(RestServlet):
+    """
+    Get members list of a room.
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)/members")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            raise NotFoundError("Room not found")
```

Comment:
Couldn't this also be from everyone having left the room?

Reply:
Yes this is possible. I think these two are not distinguishable so I edited the error message.